main/webapp/modules/core/scripts/views/data-table/menu-reconcile.js

```diff
@@ -188,6 +189,49 @@ DataTableColumnHeaderUI.extendMenu(function(column, columnHeaderUI, menu) {
     var level = DialogSystem.showDialog(frame);
   };
 
+  var doAddIdcolumn = function() {
+    var frame = $(
+      DOM.loadHTML("core", "scripts/views/data-table/add-q-column-dialog.html"));
+
+    var elmts = DOM.bind(frame);
+    elmts.dialogHeader.text($.i18n('core-views/add-id-col')+" " + column.name);
+    
+    elmts.or_views_newCol.text($.i18n('core-views/new-col-name'));
+    elmts.okButton.html($.i18n('core-buttons/ok'));
+    elmts.cancelButton.text($.i18n('core-buttons/cancel'));
+
+    var level = DialogSystem.showDialog(frame);
+    var dismiss = function() { DialogSystem.dismissUntil(level - 1); };
+
+    var o = DataTableView.sampleVisibleRows(column);
+    
+    elmts.cancelButton.click(dismiss);
+    elmts.okButton.click(function() {
+      var columnName = $.trim(elmts.columnNameInput[0].value);
+      if (!columnName.length) {
+        alert($.i18n('core-views/warning-col-name'));
+        return;
+      }
+
+      Refine.postCoreProcess(
+        "add-column", 
+        {
+          baseColumnName: column.name,  
+          newColumnName: columnName, 
+          columnInsertIndex: columnIndex + 1,
+          onError: "cell.recon.match.id"
```

Comment:
I think `onError` is meant to determine what to store when evaluating the expression gives an error, so I don't think you can have an expression here? According to the "Add column" dialog it can be one of `set-to-blank`, `store-error` or `keep-original` (which probably only makes sense when editing an existing column). I would intuitively go for `set-to-blank` here.

Reply:
Oh sorry my bad. Fixed!